Applications/FontEditor/FontEditor.cpp

```diff
@@ -263,7 +263,7 @@ FontEditorWidget::FontEditorWidget(const String& path, RefPtr<Gfx::Font>&& edite
         info_label.set_text(builder.to_string());
     };
 
-    fixed_width_checkbox.on_checked = [&](bool checked) {
+    fixed_width_checkbox.on_checked = [&, update_demo](bool checked) {
```

Comment:
This is a legitimate case, however update_demo itself also references the stack. The stack variables referenced by update_demo probably should be either capture-copied, or a better solution would be to move them to members of the class.

Reply:
And along the lines of class members, maybe `update_demo` really should just be a member function.

Reply:
```
    auto update_demo = [&] {
        demo_label_1.update();
        demo_label_2.update();
    };
```
This is not reference to the stack, as `demo_label_{1,2}` are themselves references: `GUI::Label&` to be precise.

But you're right, this is not as obvious as it should be. How could this be made more "obviously correct"?

Reply:
I'm not sure if a reference to a reference is clearly defined to do this. Are you certain that in this case it will capture those references by copying the values?

Reply:
Did a simple test and at least GCC does copy references to references implicitly for lambdas, so you're correct.

Reply:
I believe C++ does not allow references-to-references, which is one of the reasons the `T&&` syntax is even possible.

I'm not sure where in the reference to look for this, though. What do you suggest? Keep it like this? Add a `static_assert` for typechecking? Write something else entirely?

Reply:
Based on a simple test it looks like at least GCC copies the value (pointer) of the reference, which would make it work as expected. I don't know what the standard says, but it probably says the same thing then. I'd keep it like that, or make `update_demo` a member function altogether.

Reply:
In the other thread, making it a member function was a good idea. Here it would require two new members plus the function itself – nah, I think I'll stick with the lambda.

I believe that resolves it. Let me know if I missed anything :)